test/controller/check.your.answers.controller.spec.unit.ts

```diff
@@ -42,13 +45,18 @@ mockObjectionSessionMiddleware.mockImplementation((req: Request, res: Response,
   return next(new Error("No session on request"));
 });
 
+const mockGetObjection = getObjection as jest.Mock;
+
 describe("check company tests", () => {
 
   it("should render the page with company data from the session", async () => {
 
     mockGetObjectionSessionValue.mockReset();
     mockGetObjectionSessionValue.mockImplementation(() => dummyCompanyProfile);
 
+    mockGetObjection.mockReset();
```

Comment:
you can chain these together, mock.mockReset.mockImplementation
or even better use mockResolvedValue(dummyCompanyProfile) instead of implementation

Reply:
The chaining works but using mockResolvedValue(dummyCompanyProfile) doesn't. Weird because it does for the dummyObjection call. Have left both as 'mockImplementation' for consistency.

Reply:
mockGetObjection.mockReset().mockResolvedValue(dummyObjection);
works for me, make sure the param is the object and not a function

Reply:
Yep, that worked for me too, but 'mockGetObjectionSessionValue.mockReset().mockResolvedValue(dummyCompanyProfile)' doesn't work for some reason, which is why I left them the same.

Reply:
that would need a mockReturnValue/Once as it's not returning a promise.

Reply:
Oh right. Fixed now.